src/sagemaker/jumpstart/cache.py

```diff
@@ -340,32 +348,34 @@ def _retrieval_function(
                 formatted_content=model_specs
             )
         if data_type == HubContentType.MODEL:
-            info = get_info_from_hub_resource_arn(
+            hub_name, _, model_name, model_version = hub_utils.get_info_from_hub_resource_arn(
                 id_info
             )
-            hub = CuratedHub(hub_name=info.hub_name, region=info.region)
-            hub_content = hub.describe_model(
-                model_name=info.hub_content_name, model_version=info.hub_content_version
+            hub_model_description: Dict[str, Any] = self._sagemaker_session.describe_hub_content(
+                hub_name=hub_name,
+                hub_content_name=model_name,
+                hub_content_version=model_version,
+                hub_content_type=data_type
             )
+
+            model_specs = JumpStartModelSpecs(DescribeHubContentsResponse(hub_model_description), is_hub_content=True)
```

Comment:
Alternative would be to wrap the response in `Session` with `DescribeHubContentResponse`. I'd like to hear @evakravi or @akozd opinion on this

Reply:
we should do one or the other, and stick with that throughout the session module. Personally I'd prefer returning the class rather than a dictionary

Reply:
I did make all data classes for HubAPI responses. But it seems that having those as return types in `session` would cause a circular dependency as we have to import `jumpstart.types` in `session` and import `session` in `jumpstart.types`, which says it is not a good organization. I will leave the session return types as dictionary and type case somewhere else.